src/features/defi/helpers/utils.ts

```diff
@@ -34,3 +37,68 @@ export const canCoverTxFees = ({
 
   return bnOrZero(feeAssetBalanceCryptoHuman).minus(bnOrZero(estimatedGasCryptoPrecision)).gte(0)
 }
+
+type GetFeesFromFeeDataArgs = {
+  wallet: HDWallet
+  feeData: evm.FeeData
+}
+
+export const getFeesFromFeeData = async ({
+  wallet,
+  feeData: { gasLimit, gasPrice, maxFeePerGas, maxPriorityFeePerGas },
+}: GetFeesFromFeeDataArgs): Promise<evm.Fees & { gasLimit: string }> => {
+  if (!supportsETH(wallet)) throw new Error('wallet has no evm support')
```

Comment:
Noice, that will help narrowing `ETHWallet` down without having to run this one over and over everytime we get fees I reckon

Reply:
Yeah pretty much, cleans up usage and does what it needs to do